src/components/Firestore/Collection.test.tsx

```diff
@@ -249,6 +252,42 @@ it('shows the selected sub-document', async () => {
   expect(queryAllByText(/cool-doc-1/).length).toBe(expectedDocIds);
 });
 
+it('deletes collection and nested data when requested', async () => {
+  const { findByText, findByRole, queryByText } = await renderWithFirestore(
+    async (firestore) => {
+      const collectionRef = firestore.collection('whose-stuff');
+      const docRef = collectionRef.doc('cool-doc');
+      await docRef.set({ myField: 1 });
+
+      return (
+        <>
+          <Collection collection={collectionRef} />
+          <Portal />
+        </>
+      );
+    }
+  );
+
+  const menu = await findByRole('button', { name: 'Menu' });
+  await act(() => {
+    menu.click();
+    return delay(200);
+  });
+
+  confirm.mockResolvedValueOnce(true);
+
+  const deleteDocument = await findByRole('menuitem', {
+    name: 'Delete collection',
+  });
+
+  await act(() => {
+    deleteDocument.click();
+    return delay(1000);
```

Comment:
This is a bummer

Reply:
Firestore deletes are actually that slow, little I can do :(